contracts/test/LONStaking.t.sol

```diff
@@ -248,17 +249,51 @@ contract LONStakingTest is Test {
         lonStaking.stake(DEFAULT_STAKE_AMOUNT);
     }
 
-    function testStake() public {
-        BalanceSnapshot.Snapshot memory userLon = BalanceSnapshot.take(user, address(lon));
+    function _stakeAndValidate(address staker, uint256 stakeAmount) internal {
+        BalanceSnapshot.Snapshot memory stakerLon = BalanceSnapshot.take(staker, address(lon));
         BalanceSnapshot.Snapshot memory lonStakingLon = BalanceSnapshot.take(address(lonStaking), address(lon));
-        BalanceSnapshot.Snapshot memory userXLON = BalanceSnapshot.take(user, address(lonStaking));
-        uint256 stakeAmount = DEFAULT_STAKE_AMOUNT;
+        BalanceSnapshot.Snapshot memory stakerXLON = BalanceSnapshot.take(staker, address(lonStaking));
         uint256 expectedStakeAmount = _getExpectedXLON(stakeAmount);
-        vm.prank(user);
+        vm.startPrank(staker);
+        if (lon.allowance(staker, address(lonStaking)) == 0) {
+            lon.approve(address(lonStaking), type(uint256).max);
+        }
         lonStaking.stake(stakeAmount);
-        userLon.assertChange(-int256(stakeAmount));
+        stakerLon.assertChange(-int256(stakeAmount));
         lonStakingLon.assertChange(int256(stakeAmount));
-        userXLON.assertChange(int256(expectedStakeAmount));
+        stakerXLON.assertChange(int256(expectedStakeAmount));
+        vm.stopPrank();
+    }
+
+    function testStake() public {
+        _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT);
+    }
+
+    function testFuzz_Stake(uint256 stakeAmount) public {
+        vm.assume(stakeAmount > 0);
+        vm.assume(stakeAmount <= lon.cap());
+        vm.assume(stakeAmount.add(lon.totalSupply()) <= lon.cap());
+
+        lon.mint(user, stakeAmount);
+        _stakeAndValidate(user, stakeAmount);
+    }
+
+    function testFuzz_StakeMultiple(uint80[16] memory stakeAmounts) public {
+        // LON cap lies between 2**87 and 2**88, setting upper bound of stake amount to 2**80 - 1 so stake amount will not exceed cap
```

Comment:
Setting type of `stakeAmount` to `uint80` so it's max value is `2**80 - 1` so all generated numbers will be less than LON cap. This is to save time from generating invalid `stakeAmount` because now in each run we generate a total of 16 numbers instead of 1, one invalid number would result in generating 16 new numbers.